### PR TITLE
[Disability Benefits] Create tests for date fields

### LINKED ISSUE #108295: 526 'XX' Data Issue - Discovery (Frontend)
### Summary:

In certain date fields, there is a bug where in certain situations (unknown currently) a users can get an invalid date past validation and into the submission DB record.

We want to discover how and where this can occur, document it, and see if we can add additional frontend validation or logic to prevent it form occuring (in new tickets, this is discovery)

### Acceptance Criteria:

- [x] Investigate/Enumerate all places a date is used in 526 flow
- [x] Document what inputs the FE accepts for each
  - [ ] optionality
    - [ ] which portions are optional
  - [x] if it stores as a datetime object or just a string
  - [x] what validations it checks on the FE
  - [x] Document the location in the formData each date can appear
  - [ ] document any deficiencies in validation
    - [ ] Create (and link here) new FE tickets to fix/resolve discovered issues

**Previous tickets:**
https://github.com/department-of-veterans-affairs/va.gov-team/issues/106340
https://github.com/department-of-veterans-affairs/va.gov-team/issues/92417
https://github.com/department-of-veterans-affairs/va.gov-team/issues/93403
https://github.com/department-of-veterans-affairs/va.gov-team/issues/94584

Example logs:

https://vagov.ddog-gov.com/logs?query=%40named_tags.jid%3A1be4f56411240ebddc234d85&agg_m=count&agg_m_source=base&agg_t=count&clustering_pattern_field_path=message&cols=host%2Cservice&event=AwAAAZXoBZ3k3J2d8wAAABhBWlhvQmFiUkFBQVZwNEhwdDhkSXp3QkkAAAAkMDE5NWU4MGItY2NhZS00YTM4LTk5M2QtYzEyNDY0NzQzOWU1AAIhAA&fromUser=true&messageDisplay=inline&refresh_mode=sliding&storage=hot&stream_sort=desc&viz=stream&from_ts=1743187268254&to_ts=1743446468254&live=true

The locations/counts are available in the [526 dahsboard](https://vagov.ddog-gov.com/dashboard/ygg-v6d-nza/benefits-form-526-disability-compensation?fromUser=false&refresh_mode=sliding&from_ts=1744825814284&to_ts=1745430614284&live=true)

Image

### LINKED ISSUE #110067: Create and perform comprehensive Tests for All Date Fields - Phase 2
## Value Statement

***As a*** QA engineer
***I want to*** validate all date utility logic and form usage
***So that*** we catch regressions and ensure consistency across scenarios

## Objective

Achieve 100% unit test coverage for all date-related functions in `@src/applications/disability-benefits/all-claims/` to resolve the date XX bug. To cover us for the lift in shift for the new date utility module coming in.

## Description

Build unit and integration tests for the date utility and every form section using date fields.

## Implementation Notes
* Add integration tests for sections with date inputs

## Acceptance Criteria

- [ ] Unit tests exist for each utility function
- [ ] Integration tests validate form-wide usage
- [ ] Opt-out logic and partial date edge cases tested

#### `formatDate()`

- [x] Test valid dates with default DATE_FORMAT
- [x] Test valid dates with custom format
- [x] Test return 'Unknown' for invalid/null/undefined dates
- [x] Test handling of partial dates with 'X' placeholders

#### `formatDateRange()`

- [x] Test valid ranges with both dates
- [x] Test ranges with only 'from' or 'to' date
- [x] Test return 'Unknown' for empty/invalid ranges

#### `isValidFullDate()`

- [x] Test valid YYYY-MM-DD format
- [x] Test rejection of partial dates (XXXX-MM-DD, YYYY-XX-XX)
- [x] Test min/max year boundaries
- [x] Test moment parsing edge cases

#### `isNotExpired()`

- [x] Test future dates return true
- [x] Test today's date returns true
- [x] Test past dates return false
- [x] Test empty/undefined dates

#### `isValidYear()`

- [x] Test 4-digit year format
- [x] Test rejection of years < 1900 or > 3000
- [x] Test rejection of future years
- [x] Test non-numeric input

#### `isLessThan180DaysInFuture()`

- [x] Test dates within 180 days
- [x] Test rejection of dates > 180 days
- [x] Test rejection of past dates

#### `startedAfterServicePeriod()`

- [x] Test treatment dates after service start
- [x] Test partial date formats (XXXX-MM-XX, YYYY-XX-XX, YYYY-MM-XX)
- [x] Test earliest service date calculation

#### `isWithinServicePeriod()`

- [x] Test dates within service periods
- [x] Test multiple service periods
- [x] Test appropriate error messages

#### `validateSeparationDate()`

- [x] Test BDD dates (90-180 days)
- [x] Test non-BDD dates (past)
- [x] Test Reserve/Guard special cases
- [x] Test invalid formats

#### `validateAge()`

- [x] Test service start after 13th birthday
- [x] Test age calculation from DOB

#### `title10BeforeRad()`

- [x] Test separation date after activation date
- [x] Test missing/invalid dates

#### `validateTitle10StartDate()`

- [x] Test activation date for Reserve/Guard
- [x] Test date after earliest Reserve/Guard start
- [x] Test date in past

#### Helper Functions

- [x] Test `isMonthOnly()`
- [x] Test `isYearOnly()`
- [x] Test `isYearMonth()`
- [x] Test `isTreatmentBeforeService()`
- [x] Test `findEarliestServiceDate()`
- [x] Test `isInFuture()`

### Date Field Coverage

Ensure test coverage for all schema date fields:

- [x] `serviceInformation.servicePeriods[].dateRange.from/to`
- [x] `serviceInformation.reservesNationalGuardService.title10Activation.*`
- [x] `veteran.dateOfBirth`
- [x] `vaTreatmentFacilities[].treatmentDateRange.from/to`
- [x] `unemployability.*` (all date fields)
- [x] `form0781.incidents[].incidentDate`
- [x] `form0781.incidents[].unitAssignedDates.from/to`
- [x] `toxicExposure.*` (all date fields with minimumYearDateRange)

### Edge Cases

- [ ] Partial dates (X placeholders)
- [ ] Timezone handling
- [x] Leap years
- [ ] Month boundaries
- [ ] Year transitions

## Definition of Done

- [ ] 100% line coverage for date utilities
- [ ] 100% branch coverage for date validations
- [x] All tests passing
- [ ] No date XX errors in test runs
- [ ] Code reviewed and approved

### LINKED ISSUE #110377: Add form wide test coverage for dates (Phase 3)
## Description 
Expand the test suite to include complete coverage of all date fields across the form, ensuring correctness of format, validation, sanitization, and submission behavior.

### Implementation Notes

* Audit every section of the 526EZ form for fields using date inputs
* Write unit tests for utility functions covering edge cases and partial dates
* Write integration tests that simulate user interaction and submission
* Validate both toxic-exposure and general date flows (e.g. separation, employment)
* Ensure opt-out and conditional flows clean date state appropriately
### Acceptance Criteria

* [ ] Unit tests cover all utility functions and their edge cases
* [ ] Integration tests simulate real user flows for all date sections
* [ ] Tests validate cleaned payload matches expected structure
* [ ] All tests pass across devices and environments

### PR DESCRIPTION
**Pull request by freeheeling**
## Are you removing, renaming or moving a folder in this PR?
- [x] No, I'm not changing any folders (skip to TeamSites and delete the rest of this section)
- [ ] Yes, I'm removing, renaming or moving a folder

### :warning: TeamSites :warning:

Did you change site-wide styles, platform utilities or other infrastructure?
- [x] No
- [ ] Yes, and I used the [proxy-rewrite steps](https://github.com/department-of-veterans-affairs/vets-website/tree/main/src/applications/proxy-rewrite#that-sounds-normal-so-whats-the-proxy-all-about) to test the injected header scenario

## Summary

- _(Summarize the changes that have been made to the platform)_
  - Added comprehensive unit tests for all date-related functions in `@src/applications/disability-benefits/all-claims/`
- _(If bug, how to reproduce)_
  - n/a
- _(What is the solution, why is this the solution)_
  - Testing coverage was incomplete. Comprehensive test coverage will help facilitate the forthcoming lift in shift of a new date utility module.
- _(Which team do you work for, does your team own the maintenance of this component?)_
  - Disability Pathways
- _(If using a flipper, what is the end date of the flipper being required/success criteria being targeted)_
  - n/a

## Related issue(s)

- https://github.com/department-of-veterans-affairs/va.gov-team/issues/110067
- https://github.com/department-of-veterans-affairs/va.gov-team/issues/108295#issuecomment-2859104300
- [Unit Test Coverage Report: Date-Related Functions Analysis](https://github.com/department-of-veterans-affairs/va.gov-team/blob/master/products/disability/526ez/engineering_research/date_field_unit_test_coverage_report.md)

## Testing done

- _Describe what the old behavior was prior to the change_
  - Testing coverage was incomplete
- _Describe the steps required to verify your changes are working as expected_
  - each modified spec file can be run using the `yarn test:unit path/to/file` command
- _Describe the tests completed and the results_
  - 18 different `unit.spec` files, all passing

## Screenshots

n/a

## What areas of the site does it impact?

- Except for modifying what seems to be an incorrectly imported module, all changes are additions to enhance current testing

### Quality Assurance & Testing

- [x] I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [x] Linting warnings have been addressed
- [ ] Documentation has been updated ([link to documentation](#) \*if necessary)
- [ ] Screenshot of the developed feature is added
- [ ] [Accessibility testing](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/prepare-for-an-accessibility-staging-review) has been performed

### Error Handling

- [ ] Browser console contains no warnings or errors.
- [ ] Events are being sent to the appropriate logging solution
- [ ] Feature/bug has a monitor built into Datadog or Grafana (if applicable)

### Authentication

- [ ] Did you login to a local build and verify all authenticated routes work as expected with a test user

## Requested Feedback

Any and all

### CODE REVIEW

File: src/applications/disability-benefits/all-claims/pages/recentJobApplications.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 import fullSchema from 'vets-json-schema/dist/21-526EZ-ALLCLAIMS-schema.json';
-import currentOrPastDateUI from 'platform/forms-system/src/js/definitions/date';
+import currentOrPastDateUI from 'platform/forms-system/src/js/definitions/currentOrPastDate';
```
Comment (freeheeling) [inline src/applications/disability-benefits/all-claims/pages/recentJobApplications.js:2]: Changed to import what I believe to be the correct module, based on the name and usage.

**Review by TranScripted — APPROVED** [review]
:shipit:

**Review by ARPerry87 — APPROVED** [review]
Looks great!

**Review by powellkerry — APPROVED** [review]

File: src/applications/disability-benefits/all-claims/tests/pages/incidentUnitAssignment.unit.spec.jsx [hunk 2]
```diff
@@ -1,74 +1,177 @@
...
-import {
-  DefinitionTester,
-  fillDate,
-  fillData,
-} from 'platform/testing/unit/schemaform-utils';
-import { waitFor } from '@testing-library/dom';
-import { ERR_MSG_CSS_CLASS } from '../../constants';
-
+import { render } from '@testing-library/react';
+import { DefinitionTester } from '@department-of-veterans-affairs/platform-testing/schemaform-utils';
```
Comment (LinnJS) [inline src/applications/disability-benefits/all-claims/tests/pages/incidentUnitAssignment.unit.spec.jsx:4]: Thank you for upgrading from enzyme to RTL! 🙏

**Review by LinnJS — APPROVED** [review]
Absolutely wonderful work!
